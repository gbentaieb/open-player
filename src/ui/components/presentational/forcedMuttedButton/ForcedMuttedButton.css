.ForcedMuttedButton {
  position: relative;
  margin-bottom: auto;
  margin-top: auto;
  border-radius: 100%;
  background-color: rgba(0, 0, 0, 0.15);
}
