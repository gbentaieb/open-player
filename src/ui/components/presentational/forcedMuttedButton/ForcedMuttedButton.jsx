import React, { Component } from 'react';
import PropTypes from 'prop-types';
import MutedIcon from '@material-ui/icons/VolumeOff';

import Button from '../button/Button';

import style from './ForcedMuttedButton.css';

class ForcedMuttedButton extends Component {
  static propTypes = {
    videoElement: PropTypes.object.isRequired,
    forcedMuted: PropTypes.bool.isRequired,
    setForcedMuted: PropTypes.func.isRequired,
  }

  static defaultProps = {
    /* Add default props */
  }

  constructor(props) {
    super(props);
    this.onClick = this.onClick.bind(this);
  }

  onClick() {
    this.props.videoElement.muted = false;
    this.props.setForcedMuted(false);
  }

  render() {
    return this.props.forcedMuted ? (
      <div className={style.ForcedMuttedButton}>
        <Button
          onClick={this.onClick}
          Icon={MutedIcon}
        />
      </div>
    ) : null;
  }
}

export default ForcedMuttedButton;
