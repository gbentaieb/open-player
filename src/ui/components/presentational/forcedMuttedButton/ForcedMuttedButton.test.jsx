import React from 'react';
import ForcedMuttedButton from './ForcedMuttedButton';

const defaultProps = {
  videoElement: document.createElement('video'),
  forcedMuted: true,
  setForcedMuted: () => {},
};

describe('ForcedMuttedButton snapshot', () => {
  test('ForcedMuttedButton > Snapshot', () => {
    const wrapper = shallow(<ForcedMuttedButton {...defaultProps} />);
    expect(wrapper).toMatchSnapshot();
  });
});

describe('ForcedMuttedButton methods', () => {
  test('ForcedMuttedButton > onClick', () => {
    const videoElement = { muted: true };
    const setForcedMuted = jest.fn();
    const props = { ...defaultProps, videoElement, setForcedMuted };

    const wrapper = shallow(<ForcedMuttedButton {...props} />);
    const instance = wrapper.instance();
    instance.onClick();

    expect(instance.props.videoElement.muted).toBe(false);
    expect(instance.props.setForcedMuted).toHaveBeenCalledWith(false);
  });
});
