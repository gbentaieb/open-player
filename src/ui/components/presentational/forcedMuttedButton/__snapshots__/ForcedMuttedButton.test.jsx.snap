// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForcedMuttedButton snapshot ForcedMuttedButton > Snapshot 1`] = `
<div>
  <WithStyles(Button)
    Icon={[Function]}
    onClick={[Function]}
  />
</div>
`;
